tools/rewriter/GenCallAsm.cpp

```diff
@@ -1212,5 +1212,12 @@ std::string emit_asm_wrapper(AbiSignature sig,
   // for indirect wrappers or `.text` for the direct case
   add_asm_line(aw, ".previous");
 
-  return aw.ss.str();
+  std::string wrapper = "asm("s;
+  if (as_macro) {
+    wrapper += "\\";
+  }
+  wrapper += "\n";
+  wrapper += aw.ss.str();
+  wrapper += ");\n";
+  return wrapper;
```

Comment:
Why is this necessary now? It's not clear to me why your changes here require that we now manually wrap the generated asm in `asm()` when we previously could just return `aw.ss.str()` directly, but I'm also not familiar with this code.

Reply:
I just moved it from all of the many call sites.  Might not be necessary now after I changed how I was doing things, but it should be a non-functional change.

Reply:
Ah okay that's fine then, I just hadn't noticed the places where you had removed it at the call sites.